## Informe de Calidad de los datos proveniente de Yelp:
- "checkin" 

In [1]:
#Librerias a utilizar
import pandas as pd

- Leer los archivo JSON en un DataFrame de Pandas

In [2]:
checkin = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Yelp/checkin.json', lines=True)

In [3]:
# Tamaño del archivo
checkin.shape

(131930, 2)

Este archivo contiene 131930 filas y 2 columnas

In [4]:
# Vista
checkin.head()

business_id                                               date
0  ---kPU91CF4Lq2-WlRu9Lw  2020-03-13 21:10:56, 2020-06-02 22:18:06, 2020...
1  --0iUa4sNDFiZFrAdIWhZQ  2010-09-13 21:43:09, 2011-05-04 23:08:15, 2011...
2  --30_8IhuyMHbSOcNWd6DQ           2013-06-14 23:29:17, 2014-08-13 23:20:22
3  --7PUidqRWpRSpXebiyxTg  2011-02-15 17:12:00, 2011-07-28 02:46:10, 2012...
4  --7jw19RH9JKXgFohspgQw  2014-04-21 20:42:11, 2014-04-28 21:04:46, 2014...

El DF muestra dos columnas: 'business_id' y 'date'; 

La columna 'business_id' contiene un ID único para cada negocio en el conjunto de datos, mientras que la columna 'date' parece ser una lista de fechas en las que se han registrado comentarios o revisiones para cada negocio.

In [5]:
# Columnas
checkin. columns

Index(['business_id', 'date'], dtype='object')

In [7]:
# Obtener información sobre los tipos de datos, valores faltantes 
checkin.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 131930 entries, 0 to 131929
Data columns (total 2 columns):
 #   Column       Non-Null Count   Dtype 
---  ------       --------------   ----- 
 0   business_id  131930 non-null  object
 1   date         131930 non-null  object
dtypes: object(2)
memory usage: 2.0+ MB


Contiene 131930 filas y 2 columnas. Ambas columnas son de tipo object, lo que significa que contienen texto. No hay valores nulos en ninguna de las columnas.

In [8]:
# Estadísticas básicas de los datos
checkin.describe()

business_id  \
count                   131930   
unique                  131930   
top     ---kPU91CF4Lq2-WlRu9Lw   
freq                         1   

                                                     date  
count                                              131930  
unique                                             131930  
top     2020-03-13 21:10:56, 2020-06-02 22:18:06, 2020...  
freq                                                    1

El resultado de aplicar describe() a este DF solo muestra información sobre la cantidad de valores únicos y la frecuencia del valor más común (incluyendo la ID del negocio y la fecha correspondiente)

In [9]:
# Datos nulos de cada columna 
checkin.isnull().sum()

business_id    0
date           0
dtype: int64

In [10]:
# Encunetro duplicados
checkin.duplicated().sum()

0

No contiene duplicados

- "tip"

In [11]:
# Abro archivo JSON
tip = pd.read_json('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Yelp/tip.json', lines=True)

In [12]:
# Tamaño del archivo
tip.shape

(908915, 5)

Contiene 908915 filas y 5 columnas.

In [13]:
# Vista
tip.head()

user_id             business_id  \
0  AGNUgVwnZUey3gcPCJ76iw  3uLgwr0qeCNMjKenHJwPGQ   
1  NBN4MgHP9D3cw--SnauTkA  QoezRbYQncpRqyrLH6Iqjg   
2  -copOvldyKh1qr-vzkDEvw  MYoRNLb5chwjQe3c_k37Gg   
3  FjMQVZjSqY8syIO-53KFKw  hV-bABTK-glh5wj31ps_Jw   
4  ld0AperBXk1h6UbqmM80zw  _uN0OudeJ3Zl_tf6nxg5ww   

                                                text                date  \
0                     Avengers time with the ladies. 2012-05-18 02:17:21   
1  They have lots of good deserts and tasty cuban... 2013-02-05 18:35:10   
2             It's open even when you think it isn't 2013-08-18 00:56:08   
3                          Very decent fried chicken 2017-06-27 23:05:38   
4             Appetizers.. platter special for lunch 2012-10-06 19:43:09   

   compliment_count  
0                 0  
1                 0  
2                 0  
3                 0  
4                 0

El conjunto de datos representa reseñas de Yelp y contiene información como el ID del usuario que hizo la reseña, el ID del negocio al que se refiere la reseña, el texto de la reseña, la fecha de la reseña y la cantidad de elogios que recibió la reseña.

In [14]:
# Columnas
tip. columns

Index(['user_id', 'business_id', 'text', 'date', 'compliment_count'], dtype='object')

In [15]:
# Obtener información sobre los tipos de datos, valores faltantes 
tip.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 908915 entries, 0 to 908914
Data columns (total 5 columns):
 #   Column            Non-Null Count   Dtype         
---  ------            --------------   -----         
 0   user_id           908915 non-null  object        
 1   business_id       908915 non-null  object        
 2   text              908915 non-null  object        
 3   date              908915 non-null  datetime64[ns]
 4   compliment_count  908915 non-null  int64         
dtypes: datetime64[ns](1), int64(1), object(3)
memory usage: 34.7+ MB


Este es un objeto DataFrame de pandas que tiene 908915 filas y 5 columnas. Las columnas y su descripción son las siguientes:

user_id: columna de objetos que contiene identificadores de usuario.

business_id: columna de objetos que contiene identificadores de negocios.

text: columna de objetos que contiene texto relacionado con la revisión del negocio.

date: columna de fecha y hora que indica la fecha y hora en que se creó la revisión.

compliment_count: columna de enteros que indica la cantidad de cumplidos que ha recibido la revisión.

El DataFrame utiliza 34.7+ MB de memoria y tiene tipos de datos datetime64 para la columna date, int64 para la columna compliment_count y object para las columnas user_id, business_id y text.

In [16]:
# Estadísticas básicas de los datos
tip.describe()

compliment_count
count     908915.000000
mean           0.012525
std            0.120763
min            0.000000
25%            0.000000
50%            0.000000
75%            0.000000
max            6.000000

La media de la columna es de 0.012525, lo que indica que la mayoría de las revisiones no recibieron cumplidos, reflejado tambien en el mínimo. El valor máximo es 6, lo que significa que hay algunas revisiones que recibieron hasta 6 cumplidos.

El percentil 25, 50 y 75 son 0, lo que indica que el 75% de las revisiones no recibieron cumplidos y que la mayoría de las revisiones no recibieron cumplidos en absoluto. 

La desviación estándar es de 0.120763, lo que indica que los valores de la columna compliment_count están bastante agrupados alrededor de su media.

In [17]:
# Datos nulos de cada columna 
tip.isnull().sum()

user_id             0
business_id         0
text                0
date                0
compliment_count    0
dtype: int64

No posee nulos

In [18]:
# Encunetro duplicados
tip.duplicated().sum()

67

In [ ]:
Tiene 67 duplicados

- "user"

In [19]:
# Abro archivo parquet
user = pd.read_parquet('/content/drive/MyDrive/Giselle PI/Proyecto Grupal/Yelp/user.parquet')

In [20]:
# Tamaño del archivo
user.shape

(2105597, 22)

El archivo contiene 2105597 filas y 22 columnas

In [21]:
# Vista
user.head()

user_id    name  review_count        yelping_since  useful  \
0  qVc8ODYU5SZjKXVBgXdI7w  Walker           585  2007-01-25 16:47:26    7217   
1  j14WgRoU_-2ZE1aw1dXrJg  Daniel          4333  2009-01-25 04:35:42   43091   
2  2WnXYQFK0hXEoTxPtV2zvg   Steph           665  2008-07-25 10:41:00    2086   
3  SZDeASXq7o05mMNLshsdIA    Gwen           224  2005-11-29 04:38:33     512   
4  hA5lMy-EnncsH4JoR-hFGQ   Karen            79  2007-01-05 19:40:59      29   

   funny   cool                                              elite  \
0   1259   5994                                               2007   
1  13066  27281  2009,2010,2011,2012,2013,2014,2015,2016,2017,2...   
2   1010   1003                           2009,2010,2011,2012,2013   
3    330    299                                     2009,2010,2011   
4     15      7                                                      

                                             friends  fans  ...  \
0  NSCy54eWehBJyZdG2iE84w, pe42u7DcCH2QmI81NX-8qA...   267  ...   
1  ueRPE0CX75ePGMqOFVj6IQ, 52oH4DrRvzzl8wh5UXyU0A...  3138  ...   
2  LuO3Bn4f3rlhyHIaNfTlnA, j9B4XdHUhDfTKVecyWQgyA...    52  ...   
3  enx1vVPnfdNUdPho6PH_wg, 4wOcvMLtU6a9Lslggq74Vg...    28  ...   
4  PBK4q9KEEBHhFvSXCUirIw, 3FWPpM7KU1gXeOM_ZbYMbA...     1  ...   

   compliment_more  compliment_profile  compliment_cute  compliment_list  \
0               65                  55               56               18   
1              264                 184              157              251   
2               13                  10               17                3   
3                4                   1                6                2   
4                1                   0                0                0   

   compliment_note  compliment_plain  compliment_cool  compliment_funny  \
0              232               844              467               467   
1             1847              7054             3131              3131   
2               66                96              119               119   
3               12                16               26                26   
4                1                 1                0                 0   

   compliment_writer  compliment_photos  
0                239                180  
1               1521               1946  
2                 35                 18  
3                 10                  9  
4                  0                  0  

[5 rows x 22 columns]

Este es un DataFrame que contiene información de usuarios de Yelp. Cada fila representa a un usuario y las columnas incluyen información como el id del usuario (user_id), su nombre (name), la cantidad de reseñas que ha escrito (review_count), la fecha en que se unió a Yelp (yelping_since), y su número de seguidores (fans), entre otras. Algunas columnas adicionales incluyen información sobre el número de votos útiles, divertidos y geniales que ha recibido el usuario en sus reseñas, así como el número de cumplidos que ha recibido en diferentes categorías.

In [22]:
# Columnas
user. columns

Index(['user_id', 'name', 'review_count', 'yelping_since', 'useful', 'funny',
       'cool', 'elite', 'friends', 'fans', 'average_stars', 'compliment_hot',
       'compliment_more', 'compliment_profile', 'compliment_cute',
       'compliment_list', 'compliment_note', 'compliment_plain',
       'compliment_cool', 'compliment_funny', 'compliment_writer',
       'compliment_photos'],
      dtype='object')

In [23]:
# Obtener información sobre los tipos de datos, valores faltantes 
user.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2105597 entries, 0 to 2105596
Data columns (total 22 columns):
 #   Column              Dtype  
---  ------              -----  
 0   user_id             object 
 1   name                object 
 2   review_count        int64  
 3   yelping_since       object 
 4   useful              int64  
 5   funny               int64  
 6   cool                int64  
 7   elite               object 
 8   friends             object 
 9   fans                int64  
 10  average_stars       float64
 11  compliment_hot      int64  
 12  compliment_more     int64  
 13  compliment_profile  int64  
 14  compliment_cute     int64  
 15  compliment_list     int64  
 16  compliment_note     int64  
 17  compliment_plain    int64  
 18  compliment_cool     int64  
 19  compliment_funny    int64  
 20  compliment_writer   int64  
 21  compliment_photos   int64  
dtypes: float64(1), int64(16), object(5)
memory usage: 353.4+ MB


Este es otro conjunto de datos, en formato de dataframe de Pandas, tiene 22 columnas y 2105597 filas. Cada fila representa un usuario diferente y cada columna representa una característica diferente del usuario. Algunas de las características incluyen:

user_id: una cadena de caracteres que identifica de forma única a cada usuario

name: el nombre del usuario

review_count: el número de reseñas que el usuario ha escrito en Yelp

yelping_since: la fecha en que el usuario escribió su primera reseña en Yelp

useful, funny, cool: el número de votos útiles, divertidos y geniales que las reseñas del usuario han recibido

elite: una lista de años en los que el usuario ha sido miembro de Yelp Elite

friends: una lista de los amigos del usuario en Yelp

fans: el número de usuarios que han marcado al usuario como "amigo" en Yelp

average_stars: la calificación promedio de las reseñas del usuario en Yelp

compliment_ *: el número de cumplidos que el usuario ha recibido en cada una de varias categorías diferentes (como "compliment_hot" o "compliment_profile")

In [24]:
# Estadísticas básicas de los datos
user.describe()

review_count        useful         funny          cool          fans  \
count  2.105597e+06  2.105597e+06  2.105597e+06  2.105597e+06  2.105597e+06   
mean   2.846875e+01  5.516853e+01  2.267667e+01  3.143013e+01  1.913746e+00   
std    1.042146e+02  7.388068e+02  4.628811e+02  6.379274e+02  2.436510e+01   
min    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
25%    2.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
50%    6.000000e+00  3.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
75%    1.900000e+01  1.500000e+01  3.000000e+00  4.000000e+00  0.000000e+00   
max    1.747300e+04  2.062960e+05  1.858230e+05  1.998780e+05  1.249700e+04   

       average_stars  compliment_hot  compliment_more  compliment_profile  \
count   2.105597e+06    2.105597e+06     2.105597e+06        2.105597e+06   
mean    3.641863e+00    2.444596e+00     3.968618e-01        2.532631e-01   
std     1.159519e+00    7.963704e+01     1.405658e+01        1.767698e+01   
min     1.000000e+00    0.000000e+00     0.000000e+00        0.000000e+00   
25%     3.000000e+00    0.000000e+00     0.000000e+00        0.000000e+00   
50%     3.880000e+00    0.000000e+00     0.000000e+00        0.000000e+00   
75%     4.500000e+00    0.000000e+00     0.000000e+00        0.000000e+00   
max     5.000000e+00    2.578400e+04     1.350100e+04        1.418000e+04   

       compliment_cute  compliment_list  compliment_note  compliment_plain  \
count     2.105597e+06     2.105597e+06     2.105597e+06      2.105597e+06   
mean      1.858665e-01     9.631663e-02     1.921464e+00      4.124809e+00   
std       1.170051e+01     1.049796e+01     6.177753e+01      1.364953e+02   
min       0.000000e+00     0.000000e+00     0.000000e+00      0.000000e+00   
25%       0.000000e+00     0.000000e+00     0.000000e+00      0.000000e+00   
50%       0.000000e+00     0.000000e+00     0.000000e+00      0.000000e+00   
75%       0.000000e+00     0.000000e+00     0.000000e+00      0.000000e+00   
max       1.365400e+04     1.266900e+04     5.903100e+04      1.010970e+05   

       compliment_cool  compliment_funny  compliment_writer  compliment_photos  
count     2.105597e+06      2.105597e+06       2.105597e+06       2.105597e+06  
mean      3.805567e+00      3.805567e+00       1.459133e+00       1.499514e+00  
std       1.057484e+02      1.057484e+02       3.676692e+01       9.607754e+01  
min       0.000000e+00      0.000000e+00       0.000000e+00       0.000000e+00  
25%       0.000000e+00      0.000000e+00       0.000000e+00       0.000000e+00  
50%       0.000000e+00      0.000000e+00       0.000000e+00       0.000000e+00  
75%       0.000000e+00      0.000000e+00       0.000000e+00       0.000000e+00  
max       4.996700e+04      4.996700e+04       1.593400e+04       8.263000e+04

In [25]:
# Datos nulos de cada columna 
user.isnull().sum()

user_id               0
name                  0
review_count          0
yelping_since         0
useful                0
funny                 0
cool                  0
elite                 0
friends               0
fans                  0
average_stars         0
compliment_hot        0
compliment_more       0
compliment_profile    0
compliment_cute       0
compliment_list       0
compliment_note       0
compliment_plain      0
compliment_cool       0
compliment_funny      0
compliment_writer     0
compliment_photos     0
dtype: int64

No contiene nulos

In [26]:
# Encunetro duplicados
user.duplicated().sum()

117700

Contiene 117700 valores duplicados.